bookkeeper-server/src/main/java/org/apache/bookkeeper/ssl/SecurityHandlerFactory.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.bookkeeper.ssl;
+
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import java.util.Date;
+
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+public interface SecurityHandlerFactory {
```

Comment:
IMHO As this class is tied to Netty4 we should warn in the docs (javadocs of the class) that it used only internally, and discourage "custom" implementations